version: "3.7"

services:
  django:
    build:
      context: .
      dockerfile: docker/production.Dockerfile
    container_name: zhmp
    # command: python manage.py runserver 0.0.0.0:8000
    command: ["sh", "/app/docker/bin/start.sh"]
    restart: on-failure
    environment:
      - DJANGO_SETTINGS_MODULE=config.django.production
      - ALLOWED_HOSTS=*
      - CORS_ALLOWED_ORIGINS=http://10.4.55.209:8000
      - GUNICORN_LOG_LEVEL=info
      - DJANGO_DEBUG=0
    volumes:
      - .:/app
    ports:
      - "8000:8000"

  django_q:
    build:
      context: .
      dockerfile: docker/production.Dockerfile
    command: python manage.py qcluster
    environment:
     - DJANGO_SETTINGS_MODULE=config.django.production
     - DJANGO_DEBUG=0
    volumes:
      - .:/app
    restart: on-failure

  nginx:
    build:
      context: docker/nginx
      dockerfile: Dockerfile
    container_name: mynginx
    ports:
      - "88:88"
      - "82:80"
    depends_on:
      - django
    volumes:
      - ./static:/usr/share/nginx/html
      - ./media:/usr/share/nginx/media
      - /etc/localtime:/etc/localtime:ro
    logging:
      driver: "json-file"
      options:
        max-size: "200k"
        max-file: "10"